packages/ethereum/src/erc20/connector.ts

```diff
@@ -1,29 +1,34 @@
-import { ERC20abi } from "../abi/erc20.abi";
-import { OiConnector, OiConnectorConf } from "./connector";
-import { useContractConnector } from "./connectors";
-import { AssetArtifact, lookupCaipTag } from "../resolver";
-import { useABI } from "../contracts";
-import { coreApp } from "../plugin";
+import { ERC20abi } from "./abi";
+import { AssetArtifact, lookupCaipTag, useABI, useContractConnector,  OiChainTokenSupply, OiContractConnector, OiContractConnectorParams  } from "@openibex/chain";
+import { plugin } from "../plugin";
 import { AccountId } from "caip";
-import { OiChainTokenSupply } from "../primitives/supply";
 import { EventLog } from "ethers";
 
 /**
- * The OiErc20Connector is listening to the standard events of ERC20 and creates a few primitives:
+ * The OiErc20Connector is listening to the standard events of ERC20 and creates a few producers:
  * - Supply: All mints and burns, new supply and tries of all minters and burners in that block and whole chain.
  * - Transfers: Transfer volumes, all senders and recipients in that block and on chain until block.
  * - Holders: All holders in that block and on-chain as trie. As well a holders table that represents latest status.
  *
  */
-export class OiErc20 extends OiConnector {
-
-  constructor(assetArtifact: AssetArtifact, params: OiConnectorConf) {
+export class OiErc20 extends OiContractConnector {
+
+  /**
+   * Constructor.
+   * 
+   * @param assetArtifact AssetArtifact, has to be 'erc20' denominated.
+   * @param params 
+   */
+  constructor(assetArtifact: AssetArtifact, params: OiContractConnectorParams) {
     super(assetArtifact, params);
-    super.addPrimitive('Transfer', new OiChainTokenSupply(assetArtifact, params?.index));
+    super.addProducer('Transfer', new OiChainTokenSupply(assetArtifact, params?.index));

```

Comment:
This is broken. `OiChainTokenSupply`'s constructor expects `constructor(assetArtifact: AssetArtifact, namespace = pluginNamespace, plugin = pluginName)`, hence. expects the plugin name as third parameter. You pass a setting here. 


Please fix (also, why did this not cause any problems while testing?)

Reply:
This was apparently already broken in the old repository and not noticed so far. Since the constructor does not take any parameters which would be related to params, removing this parameter seems the right option. Done so in https://github.com/openibex/openibex/pull/10/commits/06d1741ee8c2d6582ac5355317e4e05f4ac8773e

Please confirm and resolve this conversation if this is correct

Reply:
Hm... I took some parameters out while documenting. Currently params are used. Definable wether caip tags get resolved or not. Was a bad idea to take the "indexed" param out, and explains why it wasnt seen by testing. (omited tests since I only "changed docs")

Reply:
We can keep the params out for now and gradually re-introduce them on demand. Scraper changes the connector orchestration anyways.